src/System.js

```diff
@@ -46,7 +47,24 @@ export class System {
         if (!Components || Components.length === 0) {
           throw new Error("'components' attribute can't be empty in a query");
         }
+
+        // Detect if the components have already been registered
+        let unregisteredComponents = Components.filter(
```

Comment:
I saw some discussion on the ecsy Discord channel where people were having Systems register their own components as a way of having singleton state. If they did this in init(), I think this will become a breaking change. Could be a non-issue if this just isn't the way things should be done. 

Another idea would be to have the `for (var queryName in this.constructor.queries)` iteration stamp queries with ids if they don't have one yet.

Reply:
Hi @zeddic I'm not sure if you are in the ecsy discord (https://discord.gg/cFnrQ2v ) so I paste here what I was asking the community for feedback:
```

15 PM] fernandojsg: @here I'd like to get feedback from the community in one change that could be a breaking change, in this PR: https://github.com/MozillaReality/ecsy/pull/222
TL;DR we will skip using Component.name for accessing components on entities, queries and systems, instead we will generate a Component._typeId when the component is registered.
That force us to register components before the systems that will use them. I added a check if you define a query of unregistered components, it will throw an error
GitHub
Fix query collision by fernandojsg · Pull Request #222 · MozillaRea...
Based on the proposal by @zeddic in #215 to fix the minification issues.

It fixes the Utils.queryKey function to use Component._typeId instead of names, so we don&#39;t rely on names anymore o...

[9:15 PM] fernandojsg: one of the uses cases that will break is registering the components directly on the System.init()
[9:17 PM] fernandojsg: I was thinking about taking the creation of the queries out of the System.constructor and then do something like 
 
system = new System()
system.init()
system._createQueries()
[9:18 PM] fernandojsg: but that could be another breaking change as people could be accessing the queries directly on the init  too
[9:18 PM] fernandojsg: can you tell me if you are using any of these two patterns currently?
```

Both alternatives will be a breaking change anyway